examples/example_diffusion.py

```diff
@@ -0,0 +1,243 @@
+    eqn = Eq(u.dt, a * (u.dx2 + u.dy2))
+    stencil = solve(eqn, u.forward)[0]
+    op = Operator(stencils=Eq(u.forward, stencil), subs={h: spacing, s: dt},
+                  nt=timesteps, shape=(nx, ny), spc_border=1, time_order=1)
+    # Execute the generated Devito stencil operator
+    tstart = time.time()
+    op.apply()
+    runtime = time.time() - tstart
+    log("Devito: Diffusion with dx=%0.4f, dy=%0.4f, executed %d timesteps in %f seconds"
+        % (spacing, spacing, timesteps, runtime))
+    return u.data[1, :], runtime
+
+
+def animate(field):
+    """Animate solution field"""
+    import matplotlib.pyplot as plt
+    xvals = np.linspace(0, 1., field.shape[0])
+    yvals = np.linspace(0, 1., field.shape[1])
+    levels = np.linspace(-.1, 1.1, 21)
+    cs = plt.contourf(xvals, yvals, field, levels=levels)
+    plt.colorbar(cs)
+    plt.show()
+
+
+def test_diffusion2d(spacing=0.01, timesteps=1000):
+    ui = ring_initial(spacing=spacing)
+    u, _ = execute_devito(ui, spacing=spacing, timesteps=timesteps)
+    assert(u.max() < 2.4)
+    assert(np.linalg.norm(u, ord=2) < 13)
+
+
+if __name__ == "__main__":
+    description = """
+Example script demosntrating several approaches to solving the
+2D diffusion equation. The various modes are pure Python,
+vectorized numpy, a lambdified SymPy equation and the Devito API.
+
+Please not that the default settings for spacing, and timesteps are
+chosen to highlight that Devito and "vectorised numpy" are faster than
+pure Python or the sympy.lambdify kernels approach.  For a fair
+performance comparison between Devito and "vectorised numpy" we
+recommend using --spacing 0.001 -t 1000.
+"""
+    parser = ArgumentParser(description=description)
+    parser.add_argument(dest='execmode', nargs='?', default='run',
+                        choices=['run', 'bench', 'plot'],
+                        help="Script mode; either 'run', 'bench' or 'plot' ")
+    parser.add_argument('-m', '--mode', nargs='+', default='devito',
+                        choices=['python', 'numpy', 'lambdify', 'devito'],
+                        help="Example modes: python, numpy, lambdify, devito")
+    parser.add_argument('-s', '--spacing', type=float, nargs='+', default=[0.01],
+                        help='Grid spacing in x and y)')
+    parser.add_argument('-t', '--timesteps', type=int, default=20,
+                        help='Number of timesteps to run')
+    parser.add_argument('--show', action='store_true',
+                        help="Show animation of the solution field")
+    parser.add_argument('-i', '--resultsdir', default='results',
+                        help='Directory containing results')
+    parser.add_argument('-o', '--plotdir', default='plots',
+                        help='Directory to store generated plots')
+
+    args = parser.parse_args()
+    if len(args.spacing) > 2:
+        raise ValueError("Too many arguments encountered for --spacing")
```

Comment:
I *think* there's an option to have the ``ArgumentParser`` automatically checking the value of a parameter.